src/libsyntax/ext/tt/transcribe.rs

```diff
@@ -282,3 +266,18 @@ fn lockstep_iter_size(tree: &quoted::TokenTree,
         TokenTree::Token(..) => LockstepIterSize::Unconstrained,
     }
 }
+
+// An Escaper escapes the syntax context with the given syntax context.
+#[derive(Debug)]
+pub struct Escaper(pub SyntaxContext);
+
+impl Folder for Escaper {
+    fn fold_ident(&mut self, mut ident: Ident) -> Ident {
```

Comment:
Overriding `fold_ident` is not necessary, overridden `new_span` applies to identifiers as well.

Reply:
Ah, I see. Why is it necessary in the `Folder` implementation for `Marker` though?

Reply:
Okay. :-)